### PR TITLE
Remove old RPM building stuff

### PR DESCRIPTION
**Pull request by howardjohn**
This is now integrated with the same code as the deb. AS a result, I
have removed the old (unused, in CI at least) RPM logic, and moved the
debian logic up and switched from `deb` specific naming to generic
naming

Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Comment by howardjohn** [comment]
@jwendell I think this isn't needed, but let me know if that is not accurate

File: tools/packaging/packaging.mk [hunk 1]
```diff
@@ -1,8 +1,195 @@
...
+
+# Note: adding --deb-systemd ${REPO_ROOT}/tools/packaging/common/istio.service will result in
+# a /etc/systemd/system/multi-user.target.wants/istio.service and auto-start. Currently not used
+# since we need configuration.
+# --iteration 1 adds a "-1" suffix to the version that didn't exist before
+${ISTIO_OUT_LINUX}/release/istio-sidecar.deb: | ${ISTIO_OUT_LINUX} deb/fpm
+${ISTIO_OUT_LINUX}/release/istio-sidecar.rpm: | ${ISTIO_OUT_LINUX} rpm/fpm
+
+# Package the sidecar rpm file.
+rpm/fpm:
```
Comment (jwendell) [inline tools/packaging/packaging.mk:52]: Are these targets duplicated with istio.mk?

**Comment by jwendell** [comment]
IIRC the initial motivation for the RPM were for people who uses RHEL/Centos to be able to install the sidecar in their machines. Is this ability lost?

File: tools/packaging/packaging.mk (hunk 1 above)
Comment (howardjohn) [reply tools/packaging/packaging.mk:52]: I just moved them from there, that file is removed now

Comment (howardjohn) [reply tools/packaging/packaging.mk:52]: Actually looks like I forgot to delete the other one

**Comment by howardjohn** [comment]
> IIRC the initial motivation for the RPM were for people who uses RHEL/Centos to be able to install the sidecar in their machines. Is this ability lost?

The `rpm/fpm` package builds a sidecar RPm. This is now included officially as part of the release: https://gcsweb.istio.io/gcs/istio-prerelease/prerelease/1.7.0-alpha.1/rpm/

File: tools/packaging/packaging.mk (hunk 1 above)
Comment (howardjohn) [reply tools/packaging/packaging.mk:52]: Fixed

File: tools/packaging/packaging.mk [hunk 2]
```diff
@@ -1,8 +1,195 @@
...
-include tools/packaging/deb/istio.mk
+# Creates the 2 packages. BUILD_WITH_CONTAINER=1 or in CI/CD (BUILD_WITH_CONTAINER=0)
+#
+# Development/manual testing:
+#    make deb          - builds debian packaging
+#    make deb/docker   - builds a test docker image
 
-# RPM/RHEL/CENTOS stuff
-include tools/packaging/rpm/rpm.mk
+deb: ${ISTIO_OUT_LINUX}/release/istio-sidecar.deb ${ISTIO_OUT_LINUX}/release/istio.deb
```
Comment (mandarjog) [inline tools/packaging/packaging.mk:10]: Is this copied verbatim from the previous deb/istio.mk ?
I have a merge conflict now with that file now.

Comment (howardjohn) [reply tools/packaging/packaging.mk:10]: Yes should be the same

**Pull request merged**